File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> hm = new HashMap<>();
        for(int i=0;i<nums.length;i++)
        {
            
                int e = target-nums[i];
                if(hm.containsKey(e))
                {
                       int ans[] = new int[2];
                    ans[0] = hm.get(e);
                    ans[1] = i;
                    return ans;
                }
                else hm.put(nums[i],i);
            }
        
    int ans[] = new int [2];
        return ans;
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int , vector<int>> m;

        for(int i=0 ; i<nums.size() ; i++)
        {
            if(m.count(target-nums[i]))
            {
                vector<int> res;
                vector<int> point = m[target-nums[i]];
                int index1 = i , index2 = point[point.size() - 1];
                res = {index1 , index2};
                return res;



            }
            m[nums[i]].push_back(i);
        }
        return {-1,-1};
        
    }
};